sway-core/src/semantic_analysis/namespace.rs

```diff
@@ -118,6 +118,20 @@ impl Namespace {
         methods
     }
 
+    pub(crate) fn copy_methods_to_type(&mut self, old_type: TypeInfo, new_type: TypeInfo) {
```

Comment:
mind adding some documentation here around what it is doing and why?

Reply:
Addressed!